meta/lib.go

```diff
@@ -94,23 +96,66 @@ func FilterKeys(domainKey string, out, in map[string]string) map[string]string {
 	return out
 }
 
-func ValidNameWithPrefix(prefix, name string) string {
+func ValidNameWithPrefix(prefix, name string, customLength ...int) string {
+	maxLength := validation.DNS1123LabelMaxLength
+	if len(customLength) != 0 {
+		maxLength = customLength[0]
+	}
 	out := fmt.Sprintf("%s-%s", prefix, name)
-	return strings.Trim(out[:min(validation.DNS1123LabelMaxLength, len(out))], "-")
+	return strings.Trim(out[:min(maxLength, len(out))], "-")
 }
 
-func ValidNameWithSuffix(name, suffix string) string {
+func ValidNameWithSuffix(name, suffix string, customLength ...int) string {
+	maxLength := validation.DNS1123LabelMaxLength
+	if len(customLength) != 0 {
+		maxLength = customLength[0]
+	}
 	out := fmt.Sprintf("%s-%s", name, suffix)
-	return strings.Trim(out[max(0, len(out)-validation.DNS1123LabelMaxLength):], "-")
+	return strings.Trim(out[max(0, len(out)-maxLength):], "-")
 }
 
-func ValidNameWithPefixNSuffix(prefix, name, suffix string) string {
-	out := strings.Trim(fmt.Sprintf("%s-%s-%s", prefix, name, suffix), "-")
+func ValidNameWithPefixNSuffix(prefix, name, suffix string, customLength ...int) string {
+	maxLength := validation.DNS1123LabelMaxLength
+	if len(customLength) != 0 {
+		maxLength = customLength[0]
+	}
+	out := fmt.Sprintf("%s-%s-%s", prefix, name, suffix)
 	n := len(out)
-	if n <= validation.DNS1123LabelMaxLength {
-		return out
+	if n <= maxLength {
+		return strings.Trim(out, "-")
+	}
+	return strings.Trim(out[:(maxLength+1)/2]+out[(n-maxLength/2):], "-")
+}
+
+func ValidCronJobNameWithPrefix(prefix, name string, customLength ...int) string {
+	maxLength := MaxCronJobLength
+	if len(customLength) != 0 {
+		maxLength = customLength[0]
+	}
+	out := fmt.Sprintf("%s-%s", prefix, name)
+	return strings.Trim(out[:min(maxLength, len(out))], "-")
+}
+
+func ValidCronJobNameWithSuffix(name, suffix string, customLength ...int) string {
+	maxLength := MaxCronJobLength
+	if len(customLength) != 0 {
+		maxLength = customLength[0]
+	}
+	out := fmt.Sprintf("%s-%s", name, suffix)
+	return strings.Trim(out[max(0, len(out)-maxLength):], "-")
```

Comment:
This should just call the other method.

`return ValidNameWithSuffix(name, suffix, MaxCronJobLength)`